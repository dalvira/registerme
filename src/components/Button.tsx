import React from 'react';
import {TouchableOpacity, Text, StyleSheet} from 'react-native';
import {colors} from '../theme/colors';

interface Props {
  testID?: string;
  label: string;
  onPress: () => void;
}

const Button = (props: Props) => {
  return (
    <TouchableOpacity
      testID={props.testID}
      style={styles.button}
      onPress={props.onPress}>
      <Text style={styles.text}>{props.label}</Text>
    </TouchableOpacity>
  );
};

export default Button;

const styles = StyleSheet.create({
  button: {
    backgroundColor: colors.blue,
    borderRadius: 40,
    height: 40,
    width: 135,
    justifyContent: 'center',
    alignItems: 'center',
  },
  text: {
    color: colors.white,
    fontSize: 16,
    fontWeight: '500',
  },
});
