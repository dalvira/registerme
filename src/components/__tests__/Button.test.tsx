import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../Button';

const mockCall = jest.fn();

it('renders correctly', () => {
  const tree = renderer
    .create(<Button label="Next" onPress={mockCall} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
