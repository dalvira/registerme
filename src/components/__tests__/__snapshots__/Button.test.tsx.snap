// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#183bdb",
      "borderRadius": 40,
      "height": 40,
      "justifyContent": "center",
      "opacity": 1,
      "width": 135,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#ffffff",
        "fontSize": 16,
        "fontWeight": "500",
      }
    }
  >
    Next
  </Text>
</View>
`;
